tests/test_lowlevel.py

```diff
@@ -204,3 +204,35 @@ def test_use_proxy_from_environment(httpbin, var, scheme):
 
         # it had actual content (not checking for SOCKS protocol for now)
         assert len(fake_proxy.handler_results[0]) > 0
+
+
+def test_redirect_rfc1808_to_non_ascii_location():
+    path = u'Ⓡéq̣̇ūéšṭš'
+    expected_path = quote(path.encode('utf8')).encode('ascii')
+
+    def redirect_resp_handler(sock):
+        consume_socket_content(sock, timeout=0.5)
+        location = u'//{0}:{1}/{2}'.format(host, port, path)
+        sock.send(
+            b'HTTP/1.1 301 Moved Permanently\r\n'
+            b'Content-Length: 0\r\n'
+            b'Location: ' + location.encode('utf8') + b'\r\n'
+            b'\r\n'
+        )
+        second_request = consume_socket_content(sock, timeout=0.5)
+
+        # currently Python3 not handling non-ASCII redirects (issue #3888)
+        assert is_py3 or second_request.startswith(b'GET /' + expected_path)
```

Comment:
Right now we should probably assert what Python 3 *does* do, even if it's wrong, and just put a note on that assertion to fix it up when we change #3888.